### DISCUSSION TITLE
subgrain size

### DISCUSSION THREAD

**Discussion by pellegrinoluca** [mtex-toolbox/mtex#738] [Ask Anything] [answered]
Dear all, 
i would to calculate the subgrains size distribution of some phases. 

Considering all dataset, I can calculate subgrains by:

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree]);
gB=[grains.innerBoundary('En','En')]
condition = gB.misorientation.angle/degree < 10;
subgrains = grains(unique(gB(condition).grainId(:)))
plot (subgrains) % attached picture1

However I'm interested only in subgrains hosted in the large crystal of attached picture1. I used different ways to do this but something went wrong, e.g.: 

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[10*degree]);
grains_big = grains(grains.area>100000)
ebsd_big = ebsd(grains_big)
[subgrains,ebsd_big.grainId] = calcGrains(ebsd_big('indexed'),'threshold',[2*degree]);
plot (subgrains) % attached picture2

Can someone halp me? 

Thank you!

![picture1](https://user-images.githubusercontent.com/78413783/106590529-b986e280-653c-11eb-8a19-1e41f2302f56.jpg)
![picture2](https://user-images.githubusercontent.com/78413783/106590532-ba1f7900-653c-11eb-8498-910d848ddcd6.jpg)

**Comment by kilir** [comment, marked as answer]
Hi,
what you see is due to the fact that you have a lot of unassigned points .
One option is to work around is mentioned here: https://mtex-toolbox.github.io/GrainReconstruction.html#11; the option `'tight'` can also be supplied with a number, 0.95 is the default.
Another option maybe better suited would to replace the non-existent points explicitly by the phase `'notIndexed'`.

Cheers,
Rüdiger

**Reply by pellegrinoluca** [reply]
Hi Rüdiger,
everything works now.

Thank you!
Cheers
Luca
